piptools/writer.py

```diff
@@ -189,17 +189,18 @@ def _iter_lines(
                 yield MESSAGE_UNSAFE_PACKAGES
 
             for ireq in unsafe_requirements:
-                req = self._format_requirement(
-                    ireq,
-                    reverse_dependencies,
-                    primary_packages,
-                    marker=markers.get(key_from_ireq(ireq)),
-                    hashes=hashes,
-                )
+                ireq_key = key_from_ireq(ireq)
                 if not self.allow_unsafe:
-                    yield comment("# {}".format(req))
+                    yield comment("# {}".format(ireq_key))
```

Comment:
If you stick `continue` here, you could dedent the following code block
```suggestion
                    yield comment("# {}".format(ireq_key))
                    continue
```

Reply:
You probably right, but I think it's not related to the issue. I'd prefer to touch these lines in a separate PR. First things first.

Reply:
And it looks subjective, is there a PEP which suggests to use:
```python
for x in z:
    if x:
        foo()
    else:
        bar()    
```

instead of

```python
for x in z:
    if x:
        foo()
        continue
    bar()    
```
? Just wondering.

Reply:
Yes, it's subjective. I don't recall this being mentioned in any PEPs.
1) It reduces the indentation which is better for readability
2) There's a similar pylint rule for `return`
   ```python
   def func():
       if cond1:
           return xxx
       else:  # pylint would emit a warning here
           return yyy
   ```
   I think it's reasonable to apply this rule to other similar structures.
3) Yes, it makes sense to do this in a separate refactoring PR to keep the diff small.